apps/src/lib/kits/maker/ui/MakerStatusOverlay.jsx

```diff
@@ -19,27 +21,44 @@ const overlayDimensionsPropTypes = {
  * when there are connection issues.
  */
 export class UnconnectedMakerStatusOverlay extends Component {
+  static propTypes = {
+    ...overlayDimensionsPropTypes,
+    isConnecting: PropTypes.bool.isRequired,
+    hasConnectionError: PropTypes.bool.isRequired,
+    handleTryAgain: PropTypes.func.isRequired,
+    handleDisableMaker: PropTypes.func,
+  };
+
   render() {
-    const {width, height, scale, isConnecting, hasConnectionError} = this.props;
+    const {width, height, scale, isConnecting, hasConnectionError,
+      handleTryAgain, handleDisableMaker} = this.props;
     const dimensions = {width, height, scale};
     if (isConnecting) {
       return <WaitingToConnect {...dimensions}/>;
     } else if (hasConnectionError) {
-      return <BoardNotFound {...dimensions}/>;
+      return (
+        <BoardNotFound
+          {...dimensions}
+          handleTryAgain={handleTryAgain}
+          handleDisableMaker={handleDisableMaker}
+        />
+      );
     }
     return null;
   }
 }
-UnconnectedMakerStatusOverlay.propTypes = {
-  ...overlayDimensionsPropTypes,
-  isConnecting: PropTypes.bool.isRequired,
-  hasConnectionError: PropTypes.bool.isRequired,
-};
 export default connect(
   state => ({
     scale: getVisualizationScale(state),
     isConnecting: isConnecting(state),
     hasConnectionError: hasConnectionError(state),
+  }),
+  () => ({
+    handleDisableMaker: () => project.toggleMakerEnabled(),
+    handleTryAgain: () => {
+      studioApp.resetButtonClick();
+      studioApp.runButtonClick();
+    },
```

Comment:
This feels like a bit of an abuse of connect. Connect is meant to match props to dispatches. In neither of these are we dispatching actions. Couldn't these just be methods on the component (or passed in by the caller)?